### ISSUE TITLE
nOAuth Mitigation?

### ISSUE THREAD

**Issue by acrogenesis** [dwyl/elixir-auth-microsoft#56]
Hello,
I just started using this package. I'm wondering if it implements validation of the subject/actor? I was reading this blog post https://www.descope.com/blog/post/noauth about the nOAuth attack and I also found this microsoft documentation on how to validate the subject and actor https://learn.microsoft.com/en-us/entra/identity-platform/claims-validation#validate-the-subject

**Comment by LuchoTurtle** [comment]
Hey there! Thank you for opening the issue and sharing the `nOAuth` possible attack vector on Microsoft-auth'd apps. It was informative :)

This package is really simple and so far it does only one thing: simplifies the login and logout process on your app registrations. You'll get the token (that you can choose to only be an `access token` or have `access token ID token` depending on how you define your app registration) and you can do whatever you want with it.

The `nOAuth` issue you've linked exists and in the documentation that you've also linked, they say to **not use `email`, `preferred_username` or `unique_name`** when the user **with the token** (that this package returns)** has access to a page or piece of data. **It's an issue with how you use the token claims to allow the user to make specific actions**.

In other words, there's no mitigation this package can have since it solely returns the token to the client (you) and you can do whatever you want with it. It's up to you to be careful and not use the aforementioned claim fields as a unique identifier to restrict users to endpoints/pages/data/whatever you want.

You can use another package to read the token JWT to see the claims, since this package doesn't have it. Though since it's getting more traction, I may change what the package effectively returns to you so you can see these claims without the use of another third-party library :)

**Comment by acrogenesis** [comment]
Thanks
